nixos/modules/services/torrent/transmission.nix

```diff
@@ -223,8 +218,11 @@ in
       };
 
       package = mkPackageOption pkgs "transmission" {
-        default = "transmission_3";
-        example = "pkgs.transmission_4";
+        default =
+          if lib.versionAtLeast config.system.stateVersion "25.11" then
+            "transmission_4"
+          else
+            "transmission_3";
```

Comment:
Nice!

Reply:
(I adjusted this slightly to move the throw so that it still gives useful error messages after the alias is removed.)